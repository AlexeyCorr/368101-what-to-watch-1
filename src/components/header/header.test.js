import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

import Header from './header.jsx';

const mock = {
  user: {
    id: 1,
    email: `email@mail.ru`,
    name: `name`,
    avatarUrl: `img/avatar.jpg`,
  },
  className: `class`,
  title: `title`,
};

Enzyme.configure({adapter: new Adapter()});

it(`renders correctly`, () => {
  const {user, className, title} = mock;
  const header = shallow(<Header
    user={user}
    className={className}
    title={title}
  />);

  expect(toJSON(header)).toMatchSnapshot();
});
