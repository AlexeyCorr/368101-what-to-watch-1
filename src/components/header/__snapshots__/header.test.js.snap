// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="page-header class"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="#"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="page-title user-page__title"
  >
    title
  </h1>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <Link
        to="/favorites"
      >
        <img
          alt="User avatar"
          height="63"
          src="https://es31-server.appspot.comimg/avatar.jpg"
          width="63"
        />
      </Link>
    </div>
  </div>
</header>
`;
